src/libraries/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/HashAlgorithm.cs

```diff
@@ -99,24 +99,8 @@ public byte[] ComputeHash(Stream inputStream)
             if (_disposed)
                 throw new ObjectDisposedException(null);
 
-            // Use ArrayPool.Shared instead of CryptoPool because the array is passed out.
-            byte[] buffer = ArrayPool<byte>.Shared.Rent(4096);
-
-            int bytesRead;
-            int clearLimit = 0;
-
-            while ((bytesRead = inputStream.Read(buffer, 0, buffer.Length)) > 0)
-            {
-                if (bytesRead > clearLimit)
-                {
-                    clearLimit = bytesRead;
-                }
-
-                HashCore(buffer, 0, bytesRead);
-            }
+            inputStream.CopyTo((span, state) => ((HashAlgorithm)state!).HashCore(span), this, 4096);
 
-            CryptographicOperations.ZeroMemory(buffer.AsSpan(0, clearLimit));
```

Comment:
If a temporary buffer gets used now as part of, for example, CopyTo delegating to the base implementation, this clearing is no longer going to happen.  @bartonjs will need to comment on whether that's acceptable.

Reply:
Oh! You are right. If it is not acceptable, then we just have to close this PR 😢

Reply:
If all of the builtin implementations are stackalloc-or-nothing, then maybe it's OK.  If any of our implementations ever rents from a pool, it's unacceptable.

Reply:
@bartonjs, built-in implementations of what? Why are we concerned about zero'ing out the data read from an arbitrary stream for this one consumptive purpose?

Reply:
> built-in implementations of what?

The new CopyTo. It looks like the base implementation has this loop in it, and only MemoryStream is avoiding extra work.

So before this PR, running SHA256.ComputeHash(FileStream) and FileStream.ReadToEnd() have the same amount of GC heap data residuals (namely, whatever residuals FileStream leaves).  After this PR there's an extra residual of the last block of data processed in the CopyTo loop, and it's rooted (and retrievable) via the ArrayPool.

While one could argue that hashing is generally done on public data (ciphertext or non-encrypted data) it's still the intent of crypto code to not cause any residual data copies.  So unless we're changing FileStream, NetworkStream, et al to use their existing buffers for Span-based CopyTo, I want to not take this change here.

Reply:
> It looks like the base implementation has this loop in it

Right

> only MemoryStream is avoiding extra work

Currently MemoryStream, UnmanagedMemoryStream, and Stream.Null override it.

> So unless we're changing FileStream, NetworkStream, et al to use their existing buffers for Span-based CopyTo

We potentially could for FileStream. NetworkStream doesn't have a buffer.

> I want to not take this change here.

Ok.  I disagree that this one more residual copy of the data is a problem, but I won't fight it in this case.

@manandre, thank you for the contribution.  Given the discussion, though, I'm going to close this.